pandas/io/sql.py

```diff
@@ -69,6 +72,14 @@
 # -- Helper functions
 
 
+def _cleanup_after_generator(generator, exit_stack: ExitStack):
```

Comment:
Might have slightly preferred if `_query_iterator` used `self.exit_stack` directly in the functions, but could be a follow up e.g.

```
def _query_iterator(...):
    with self.exit_stack():
        ...
```